Switching to MariaDB

Til7701 opened the pull request:
The vector store can now save vectors with up to 65532 dimensions. https://mariadb.com/docs/server/reference/data-types/numeric-data-types/vector

 - PostgreSQL is replaced with MariaDB
 - The VectorStoreExtension has been implemented for MariaDB. The implementation has been copied from the PgVectorStoreExtension, since they are very similar.
 - The table name is set based on the given model. Here only the property names were changed.
 - application YAMLs have been updated

Til7701 committed:
Switching to MariaDB

The vector store can now save vectors with up to 65532 dimensions. https://mariadb.com/docs/server/reference/data-types/numeric-data-types/vector

 - PostgreSQL is replaced with MariaDB
 - The VectorStoreExtension has been implemented for MariaDB. The implementation has been copied from the PgVectorStoreExtension, since they are very similar.
 - The table name is set based on the given model. Here only the property names were changed.
 - application YAMLs have been updated

JayPi4c commented on src/main/java/de/uol/pgdoener/civicsage/business/embedding/PgVectorStoreExtension.java:
Can't we support both, `PGVectorStore` and `MariaDBVectorStore`? Depending on the Usecase, the limit dimension size might be acceptable? Configuring which to be used can be done in spring using the appropriate annotations. 

Well, maybe not officially support, but make it work and let the user decide based on custom preferences?

JayPi4c replied:
We could still say, that we will only test and officially support MariaDB but we dont have to delete all pgsql related stuff.

Til7701 committed:
Allow switching between mariadb and postgresql as vector stores.

Added properties and configuration. The table name is now built after environment processing, since the property is now more under our control.

Til7701 committed:
Fix vector store name

Til7701 committed:
Merge remote-tracking branch 'origin/dev' into 115-switch-vector-database-to-support-larger-vectors

JayPi4c commented:
I don't know, if it is some strange setup on my side, but I happen to not get a score, or to be precise, I only get null, when using the `MariaDB` VectorStore:

```json
  {
    "startup_document": true,
    "documentId": "67651af2-3087-4a02-a84b-a482cca4f92a",
    "fileName": "some.pdf",
    "fileId": "24bba263-17dc-4f66-be1d-be1ee1e1a7f9",
    "url": null,
    "title": "...",
    "text": "...",
    "score": null
  }
```

Til7701 commented:
Good question. I am also getting `null`

Til7701 committed:
The MariaDBVectorStore does not return a score, but it returns a distance hidden in the metadata. In the case there is no score, that metadata field is read and the score is calculated from the distance.
